core/src/main/java/org/bitcoinj/core/Wallet.java

```diff
@@ -1817,6 +1842,9 @@ private void receive(Transaction tx, StoredBlock block, BlockChain.NewBlockType
             log.info("  <-pending");
 
         if (bestChain) {
+            boolean wasDead = dead.remove(txHash) != null;
```

Comment:
I disagree. I know miners "in general" remove a tx from their memory pool if a tx in conflict with it is included in a block in the best chain. 
The case I am dealing here is what happens if a miner does not remove that tx in conflict from its memory pool (because she haven't received the new best block yet or she is doing an attack or some other reason) and mines a competitive block including the tx considered dead by the rest of the network and then another block on top of it. In that scenario, the network will accept this 2 new blocks as part of the best chain, the original best block will be considered a fork, the tx that is in the fork will be dead and the tx that was dead will be considered valid.
Am I missing something here?